homeassistant/components/portainer/sensor.py

```diff
@@ -0,0 +1,140 @@
+                    ):
+                        _LOGGER.info("Deleting: %s", entity_entry.entity_id)
+                        entity_reg.async_remove(entity_entry.entity_id)
+
+    def _container_listener() -> None:
+        """Listen for changes in containers and add/remove entities dynamically."""
+        new_entities: list[ContainerSensorEntity] = []
+        for node_id in coordinator.data:
+            # Convert node_id to str if necessary, to match the expected key type in current_containers
+            node_id_str = str(node_id)
+
+            # Fetch container IDs from the latest snapshot
+            container_ids = set(
+                coordinator.data[node_id]
+                .snapshots[-1]
+                .docker_snapshot_raw.containers.keys()
+            )
+
+            # Get the current set of containers for this node, defaulting to an empty set if not present
+            current_set: set[str] = current_containers.get(node_id_str, set())
+
+            # Determine which containers were added or removed
+            added_containers = container_ids - current_set
+            removed_containers = current_set - container_ids
+
+            # Add new entities for any newly detected containers
+            if added_containers:
+                new_entities.extend(
+                    ContainerSensorEntity(
+                        coordinator, node_id, container_id, description
+                    )
+                    for container_id in added_containers
+                    for description in CONTAINER_SENSOR_TYPES
+                    if description.exists_fn(coordinator.data[node_id])
+                )
+                current_set.update(added_containers)
+
+            # Remove entities for containers that were removed
+            if removed_containers:
+                _remove_containers(removed_containers, node_id)
+
+            # Update current containers for this node
+            current_containers[node_id_str] = current_set
+
+        # Add any newly detected entities
+        if new_entities:
+            async_add_entities(new_entities)
+
+    coordinator.async_add_listener(_container_listener)
+    _container_listener()
+
+
+class ContainerSensorEntity(ContainerBaseEntity, SensorEntity):
+    """Defining the Portainer Sensors with ContainerSensorEntityDescription."""
+
+    entity_description: ContainerSensorEntityDescription
+
+    def __init__(
+        self,
+        coordinator: PortainerDataUpdateCoordinator,
+        node_id: int,
+        container_id: str,
+        description: ContainerSensorEntityDescription,
+    ) -> None:
+        """Set up ContainerSensors."""
+        super().__init__(coordinator, node_id, container_id)
+        self.entity_description = description
+        # We use container name as unique id, because the container_id changes
+        # with an update of the container
+        self._attr_unique_id = f"{node_id}-{self.container_id}-{description.key}"
```

Comment:
```suggestion
        self._attr_unique_id = f"{node_id}-{container_id}-{description.key}"
```
Is that comment correct?